webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -0,0 +1,76 @@
+import URI from 'urijs';
+import API from '../../API';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  TRIGGERS,
+} from './AutoCompleteConstants';
+
+export const getResults =
+(path, searchQuery, controller, trigger) => (dispatch) => {
+  const loadNextResults = trigger === TRIGGERS.ITEM_SELECT ? '+' : '';
+  const url = `${path}${URI.encodeQuery(searchQuery)}${loadNextResults}`;
+
+  // Starting a request
+  dispatch({
+    type: AUTO_COMPLETE_REQUEST,
+    payload: {
+      controller,
+      searchQuery,
+      status: STATUS.PENDING,
+      trigger,
+    },
+  });
+
+  // Making an API call
+  return API.get(url)
+    .then(({ data }) => {
+      const { error } = data[0] || {};
+      // The API can return errors too.
+      if (error) {
+        throw error;
+      }
+      return dispatch({
+        type: AUTO_COMPLETE_SUCCESS,
+        payload: {
+          controller,
+          results: data,
+          searchQuery,
```

Comment:
I believe data has `searchQuery` in its body.

Reply:
in other words, I believe we can use [`ajaxRequestAction`](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/redux/actions/common/index.js#L3) here.

Reply:
The data returned from the request is the results in array, e.g:
![selection_050](https://user-images.githubusercontent.com/26363699/44723810-b5d41380-aad9-11e8-99e7-7968c1b66b28.png)
That's why I needed to make some adjustments that the `ajaxRequestAction` doesn't support, 
I also needed to handle API errors that are resolved after the API request.

> I believe data has searchQuery in its body.

not sure that I understand what did you mean.

Reply:
My bad - I thought it returns `search`, `total`, and etc like in `api/v2/`